### PR TITLE
 Implement hex human-readable serde for PublicKey 

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

File: src/key.rs [hunk 1]
```diff
@@ -354,17 +366,66 @@ impl From<ffi::PublicKey> for PublicKey {
 #[cfg(feature = "serde")]
 impl ::serde::Serialize for PublicKey {
     fn serialize<S: ::serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
-        s.serialize_bytes(&self.serialize())
+        if s.is_human_readable() {
+            s.collect_str(self)
+        } else {
+			s.serialize_bytes(&self.serialize())
```
Comment (apoelstra) [inline src/key.rs]: spacing nit

**Comment by stevenroose** [comment]
Something strange happened to the last build.. Should I restart?

**Comment by apoelstra** [comment]
Yeah, I'd just kick it

**Comment by stevenroose** [comment]
Pushed spacing nits, replaced tabs with spaces.

**Comment by stevenroose** [comment]
I don't know much about rust-secp internals. Should I be worried about those 3 builds timing out? There doesn't seem to be any useful information in the logs..

**Comment by elichai** [comment]
It didn't even start to build: https://api.travis-ci.org/v3/job/562661011/log.txt
Probably a problem with windows builds in travis

**Pull request merged**